packages/ethapi/src/ethapi.ts

```diff
@@ -136,8 +136,12 @@ export class EthApi {
     let signature: Hex;
     try {
       signature = (await this.rpc("personal_sign", messageHex, signerAccount, "")).result;
-    } catch {
-      signature = (await this.rpc("eth_sign", signerAccount, messageHex)).result;
+    } catch (e) {
+      if (e.message === "Error: MetaMask Message Signature: User denied message signature.") {
+        throw e; // rethrow the metamask error to be handled in ui
```

Comment:
:ok_hand:

Reply:
Though shame/weird there's no code or anything that might be more stable than this message.

Reply:
yeah